### PR TITLE
MAINT: eliminate some pytest warnings

### LINKED ISSUE #5329: Lots of warnings in travis from scipy sparse matrix test
If you've been following travis recently, we're getting a lot of warnings about numpy matrix vs. ndarray... there's over 20,000 of them.

The cause is one test (I think):

https://github.com/statsmodels/statsmodels/blob/ec672ec27dacb9778d524d564e50eb45306c63c0/statsmodels/stats/tests/test_corrpsd.py#L309-L339

### PR DESCRIPTION
**Pull request by thequackdaddy**
Closes #5329

I also added a `-nv` option on the `wget miniconda.sh` because it shrinks the log file a bunch, and that should make it so that the docbuild will take under 10,000 lines, so you can see the entire log on travis without going to the raw log.

### CODE REVIEW

**Comment by bashtage** [comment]
Should ideally silence this stuff -- about 1k lines and not really human readable.

```
----------- coverage: platform linux, python 3.7.0-final-0 -----------
Name                                                                         Stmts   Miss Branch BrPart  Cover   Missing
------------------------------------------------------------------------------------------------------------------------
statsmodels/__init__.py                                                         39     17     10      2    49%   19-27, 35, 40-51, 18->19, 34->35
```

**Comment by bashtage** [comment]
I think this needs to add  `--cov-report= ` to the coverage options.

**Comment by bashtage** [comment]
Git hub is having some issues.  About the attr check,

`Comment to explain this change?  Appears to be panda/numpy interop.  Would it be better to be explicit (isinstance) rather than look for attribute that may not be from a DataFrame or Series?`

**Comment by bashtage** [comment]
Git hub is having some issues.  About the attr check,

`Comment to explain this change?  Appears to be panda/numpy interop.  Would it be better to be explicit (isinstance) rather than look for attribute that may not be from a DataFrame or Series?`

**Comment by bashtage** [comment]
Git hub is having some issues.  About the attr check,

`Comment to explain this change?  Appears to be panda/numpy interop.  Would it be better to be explicit (isinstance) rather than look for attribute that may not be from a DataFrame or Series?`

**Comment by thequackdaddy** [comment]
@bashtage I think github is back. I can revert, but the explanation is above. As a simpler alternative, the test case could be built with a unique index.

File: statsmodels/base/data.py [hunk 1]
```diff
@@ -242,6 +242,7 @@ def handle_missing(cls, endog, exog, missing, **kwargs):
                     raise ValueError("Shape mismatch between endog/exog "
                                      "and extra arrays given to model.")
                 # for going back and updated endog/exog
+                nan_mask = getattr(nan_mask, 'values', nan_mask)
```
[the remark this answers is not in the document]
Comment (thequackdaddy) [reply statsmodels/base/data.py]: Explained below... not in love with this, but open for ideas.

File: setup.cfg [hunk 2]
```diff
@@ -135,12 +136,10 @@ select=
     # F722: syntax error in forward type annotation
     F831,
     # F831: duplicate argument name in function definition
-    F901,
+    # F901,
     # F901: raise NotImplemented should be raise NotImplementedError
-    E306,
+    # E306,
```
[the remark this answers is not in the document]
Comment (thequackdaddy) [reply setup.cfg]: flake8 updated to 3.6.0 and this is all broken now.

File: lint.sh [hunk 3]
```diff
@@ -16,7 +16,7 @@ if [ "$LINT" == true ]; then
     # Run with --isolated to ignore config files, the files included here
     # pass _all_ flake8 checks
     echo "Linting known clean files with strict rules"
-    flake8 --isolated \
+    flake8 --isolated --ignore=E123,E126,E226,E241,W503,W504,W605 \
```
[the remark this answers is not in the document]
Comment (thequackdaddy) [reply lint.sh]: I guess that's fair, but note that flake8 updated to 3.6.0 just a few days ago, and a lot of these files now fail. Likely only a few may survive the purge.

File: statsmodels/stats/libqsturng/tests/test_qsturng.py [hunk 4]
```diff
@@ -183,6 +183,7 @@ def test_handful_to_known_values(self):
         for p,r,v,q in cases:
             assert_almost_equal(1.-p, psturng(q,r,v), 5)
 
+    @pytest.mark.xfail(raises=UnboundLocalError)
```
[the remark this answers is not in the document]
Comment (thequackdaddy) [reply statsmodels/stats/libqsturng/tests/test_qsturng.py]: Not really. The fail is pretty random. See #5336

[the remark this answers is not in the document]
Comment (jbrockmendel) [reply statsmodels/stats/libqsturng/tests/test_qsturng.py]: The try/except/print in 5336 looks custom-made to cause a UnboundLocalError.  Doesn't seem like it should be this PR's problem.

File: lint.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply lint.sh]: I'll take a look at the 3.6.0 changes, but we really ought to keep the "strict" check strict.

File: setup.cfg (hunk 2 above)
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply setup.cfg]: Some of these are solved by #5204.

[the remark this answers is not in the document]
Comment (jbrockmendel) [reply setup.cfg]: More generally, let's see if we can fix these rather than loosening standards.

[the remark this answers is not in the document]
Comment (jbrockmendel) [reply setup.cfg]: I see a lot of W605s, exactly 1 E306, and 1 F901.  All of the formerly-strict failures are W605.  Should be easy fixes.

File: lint.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (thequackdaddy) [reply lint.sh]: Much of it seems related to bad/lazy latex coding, i.e., typing `"\sum endog"`. You either need to prefix the string with an `r` to make it a raw string so latex compiles correctly, or you need to do `"\\sum endog"`.

File: statsmodels/base/data.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (thequackdaddy) [reply statsmodels/base/data.py]: On further reflection, I'll revert this.

File: lint.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply lint.sh]: I get censored around here when I express opinions that any part of statsmodels could be imperfect.  #5252 and #5353 make a dent in these.

File: statsmodels/stats/libqsturng/tests/test_qsturng.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply statsmodels/stats/libqsturng/tests/test_qsturng.py]: Should be revertable now that 5336 is closed

File: setup.cfg [hunk 5]
```diff
@@ -135,12 +136,10 @@ select=
     # F722: syntax error in forward type annotation
     F831,
     # F831: duplicate argument name in function definition
-    F901,
+    # F901,
```
Comment (jbrockmendel) [inline setup.cfg]: should be revertable following #5352

File: statsmodels/sandbox/distributions/tests/test_multivariate.py [hunk 6]
```diff
@@ -71,9 +70,9 @@ def test_mvn_mvt_3(self):
         probmvt_R = 0.9522146
         quadkwds = {'epsabs': 1e-08}
         probmvt = mvstdtprob(a2, b, corr2, df, quadkwds=quadkwds)
-        assert_almost_equal(probmvt_R, probmvt, 4)
+        assert_allclose(probmvt_R, probmvt, atol=0.0005)
         probmvn = mvstdnormcdf(a2, b, corr2, maxpts=100000, abseps=1e-5)
-        assert_almost_equal(probmvn_R, probmvn, 4)
+        assert_allclose(probmvn_R, probmvn, atol=0.0001)
```
Comment (jbrockmendel) [inline statsmodels/sandbox/distributions/tests/test_multivariate.py]: can you write this as 1e-4, above as 5e-4?

Comment (thequackdaddy) [reply statsmodels/sandbox/distributions/tests/test_multivariate.py]: Changed.

**Comment by thequackdaddy** [comment]
@josef-pkt Any comments? This will fix travis.

**Comment by josef-pkt** [comment]
no comments
I haven't caught up with pep-8 enforcement yet.

I will just merge, mainly to get the "ignore" in.

Thanks all

**Comment by jbrockmendel** [comment]
Main comment: instead of lowering standards, lets fix the underlying problem.

**Pull request merged**